scripts/teamd_increase_retry_count.py

```diff
@@ -0,0 +1,236 @@
+
+def getPortChannelConfig(portChannelName):
+    process = subprocess.run(["teamdctl", portChannelName, "state", "dump"], capture_output=True)
+    return json.loads(process.stdout)
+
+def getLldpNeighbors():
+    process = subprocess.run(["lldpctl", "-f", "json"], capture_output=True)
+    return json.loads(process.stdout)
+
+def craftLacpPacket(portChannelConfig, portName, isResetPacket=False, newVersion=True):
+    portConfig = portChannelConfig["ports"][portName]
+    actorConfig = portConfig["runner"]["actor_lacpdu_info"]
+    partnerConfig = portConfig["runner"]["partner_lacpdu_info"]
+    l2 = Ether(dst="01:80:c2:00:00:02", src=portConfig["ifinfo"]["dev_addr"], type=0x8809)
+    l3 = scapy.contrib.lacp.SlowProtocol(subtype=0x01) 
+    l4 = LACPRetryCount()
+    if newVersion:
+        l4.version = 0xf1
+    else:
+        l4.version = 0x1
+    l4.actor_system_priority = actorConfig["system_priority"]
+    l4.actor_system = actorConfig["system"]
+    l4.actor_key = actorConfig["key"]
+    l4.actor_port_priority = actorConfig["port_priority"]
+    l4.actor_port_number = actorConfig["port"]
+    l4.actor_state = actorConfig["state"]
+    l4.partner_system_priority = partnerConfig["system_priority"]
+    l4.partner_system = partnerConfig["system"]
+    l4.partner_key = partnerConfig["key"]
+    l4.partner_port_priority = partnerConfig["port_priority"]
+    l4.partner_port_number = partnerConfig["port"]
+    l4.partner_state = partnerConfig["state"]
+    if newVersion:
+        l4.actor_retry_count = 5 if not isResetPacket else 3
+        l4.partner_retry_count = 3
+    packet = l2 / l3 / l4
+    return packet
+
+def getPortChannels():
+    configDb = DBConnector("CONFIG_DB", 0)
+    portchannelTable = Table(configDb, "PORTCHANNEL")
+    return list(portchannelTable.getKeys())
+
+class LacpPacketListenThread(Thread):
+    def __init__(self, port, targetMacAddress, sendReadyEvent):
+        Thread.__init__(self)
+        self.port = port
+        self.targetMacAddress = targetMacAddress
+        self.sendReadyEvent = sendReadyEvent
+        self.detectedNewVersion = False
+
+    def lacpPacketCallback(self, pkt):
+        if pkt["LACPRetryCount"].version == 0xf1:
+            self.detectedNewVersion = True
+        return self.detectedNewVersion
+
+    def run(self):
+        sniff(stop_filter=self.lacpPacketCallback, iface=self.port, filter="ether proto 0x8809 and ether src {}".format(self.targetMacAddress),
+                store=0, timeout=30, started_callback=self.sendReadyEvent.set)
+
+def sendLacpPackets(packets, revertPackets):
+    while not revertTeamdRetryCountChanges:
+        for port, packet in packets:
+            sendp(packet, iface=port)
+        time.sleep(15)
+    if revertTeamdRetryCountChanges:
+        for port, packet in revertPackets:
+            sendp(packet, iface=port)
+
+def abortTeamdChanges(signum, frame):
+    print("Got signal {}, reverting teamd retry count change".format(signum))
```

Comment:
Is the `print` usage intentional? I think we should use `logger`.

Reply:
Switched to logger, with some messages still being explicitly being printed to console (provided the minimum log level is met). This is so that it's easy for someone to run just this script and see if it passed or not.